packages/carequality-sdk/src/client/carequality.ts

```diff
@@ -0,0 +1,147 @@
+   * @param options         Optional parameters
+   * @param options.timeout Connection timeout in milliseconds, default 120 seconds.
+   */
+  constructor(apiKey: string, apiMode: string, options: { timeout?: number } = {}) {
+    let baseUrl;
+
+    switch (apiMode) {
+      case APIMode.dev:
+        baseUrl = Carequality.devUrl;
+        break;
+      case APIMode.staging:
+        baseUrl = Carequality.stagingUrl;
+        break;
+      default:
+        throw new Error("API mode not supported.");
+    }
+
+    this.api = axios.create({
+      timeout: options?.timeout ?? DEFAULT_AXIOS_TIMEOUT,
+      baseURL: baseUrl,
+    });
+    this.apiKey = apiKey;
+  }
+
+  buildHeaders(headers?: Record<string, string>): Record<string, string> {
+    return {
+      ...headers,
+      "Accept-Encoding": "gzip",
+    };
+  }
+
+  async sendGetRequest(url: string, headers?: Record<string, string>): Promise<AxiosResponse> {
+    return this.api.get(url, { headers: this.buildHeaders(headers) });
+  }
+
+  /**
+   * Lists the indicated number of organizations. Mostly used for testing purposes.
+   * @param count Optional, number of organizations to fetch.
+   * @param start Optional, the index of the directory to start querying from; optional
+   * @returns
+   */
+  async listOrganizations({
+    count = MAX_COUNT,
+    start = 0,
+  }: {
+    count?: number;
+    start?: number;
+  }): Promise<OrganizationList> {
+    if (count < 1 || count > MAX_COUNT)
+      throw new BadRequestError(
+        `Count value must be between 1 and ${MAX_COUNT}. If you need more, use listAllOrganizations()`
+      );
+
+    const url = `${Carequality.ORG_ENDPOINT}?apikey=${this.apiKey}&${JSON_FORMAT}&_count=${count}&_start=${start}`;
+    const resp = await this.sendGetRequest(url, { "Content-Type": "application/json" });
+
+    const bundle: STU3Bundle = stu3BundleSchema.parse(resp.data.Bundle);
+    const orgs = bundle.entry.map(e => e.resource.Organization);
+    return {
+      count: orgs.length,
+      organizations: orgs,
+    };
+  }
+
+  /**
+   * Lists the whole Carequality directory.
+   *
+   * @returns organizations list and count
+   */
+  async listAllOrganizations(failGracefully = false): Promise<OrganizationList> {
+    let currentPosition = 0;
+    const results: OrganizationList = {
+      count: 0,
+      organizations: [],
+    };
+    let isDone = false;
+
+    while (!isDone) {
+      try {
+        console.log(
+          `Querying the next ${MAX_COUNT} organizations, starting from ${currentPosition}`
+        );
+        const resp = await this.listOrganizations({ start: currentPosition });
+        results.organizations.push(...resp.organizations);
+        currentPosition += MAX_COUNT;
+
+        const bundleEntryLength = resp.count;
+        console.log(
+          `Received: ${bundleEntryLength} orgs. Continuing: ${!(bundleEntryLength < MAX_COUNT)}`
+        );
+        if (bundleEntryLength < MAX_COUNT) {
```

Comment:
There was another thread about this, but essentially we're using one field for count while we have the actual list of items on another field.

I'd rely on the actual list, even though they should match, but better safe than sorry :)